examples/platforms/nrf528xx/src/radio.c

```diff
@@ -191,6 +204,57 @@ static inline void clearPendingEvents(void)
     } while (__STREXW(pendingEvents, (uint32_t *)&sPendingEvents));
 }
 
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+static void txAckProcessSecurity(uint8_t *p_ack_frame)
+{
+    otRadioFrame     ackFrame;
+    struct otMacKey *key = NULL;
+    uint8_t          keyId;
+
+    sAckedWithSecEnhAck = false;
+    otEXPECT(p_ack_frame[SECURITY_ENABLED_OFFSET] & SECURITY_ENABLED_BIT);
+
+    memset(&ackFrame, 0, sizeof(ackFrame));
+    ackFrame.mPsdu   = &p_ack_frame[1];
+    ackFrame.mLength = p_ack_frame[0];
+
+    keyId = otMacFrameGetKeyId(&ackFrame);
+
+    otEXPECT(otMacFrameIsKeyIdMode1(&ackFrame) && keyId != 0);
+
+    if (keyId == sKeyId)
+    {
+        key = &sCurrKey;
+    }
+    else if (keyId == sKeyId - 1)
+    {
+        key = &sPrevKey;
+    }
+    else if (keyId == sKeyId + 1)
+    {
+        key = &sNextKey;
+    }
+    else
+    {
+        otEXPECT(false);
```

Comment:
Add log here, or assert(false)? It seems there is a logic error if this happens.

Reply:
As discussed, it is possible for a node to received a frame with "bad" key index. It should just drop it. This implementation follows `Mac::ProcessReceiveSecurity`.

>From 15.4-2015 section 7.3.3
If the Security Enabled field in the Frame Control field is set to one, the Enh-Ack frame is protected using the procedure described in 9.2.1. The SecurityLevel, KeyIdMode, KeySource and KeyIndex shall be set to match the corresponding fields of the frame that is being acknowedged.

>From Thread 1.2 section 7.1.5
Typically, three keys are stored ... This storage of multiple keys allows migration to a new key while still retaining the old key.

>From 15.4-2015 section 9.2.1
c)Obtain KeyDescriptor. The procedure shall obtain the  KeyDescriptor using the KeyDescriptor lookup procedure as described in 9.2.2 using the KeyIdMode, KeyIndex, and KeySource with the DeviceAddressingMode  set  to  Destination  Addressing  Mode  field, DevicePanId  set  to  theDestination PAN ID field, and DeviceAddress set to the Destination Address field. If that procedure fails, the procedure shall return with a Status of UNAVAILABLE_KEY.

So if key could not be found, stack will escape the encryption procedure and the frame will be sent out unencrypted.